### DISCUSSION TITLE
INCREASED RISK OF VOLATILITY

### DISCUSSION THREAD

**Discussion by mark-leonard-reynolds** [amberelectric/public-api#202] [Q&A] [answered]
Can someone please point me to an API call/response which will allow me to see that there is a INCREASED RISK OF VOLATILITY for the day?

I can see from the live price page that the current period seems to have this info tagged.

The reason I'd like to detect this, is that I can apply a more aggresive grid battery charging policy, to ensure the battery is fully charged so that I can discharge it into likely spikes later in the day.

I can see "spikeStatus": "none",  for the current period, which of course, is not what I'm after. Battery needs time to charge up!

And this of course, is likely needing grid charge, as when the supply seems to be VOLATILE, it's often cloudy and raining or stormy, so not much sun to charge up the battery.

**Comment by madpilot** [comment, marked as answer]
Look at the spike status in the forecasts.

If you are using the `/prices/current` endpoint, you can add `?next=24` to the URL, and you'll get the next 12 hours of forecasts. They have a `spikeStatus` field. Then you can work out whether you want to pre-charge your battery or not.

**Reply by mark-leonard-reynolds** [reply]
Wow! Thanks! Much quicker than reverse engineering amber live prices page :-)

**Reply by mark-leonard-reynolds** [reply]
So, something like this :
curl -sX 'GET'  'https://api.amber.com.au/v1/sites/[insert_your_site_here]/prices/current?next=24'  -H 'accept: application/json'  -H 'Authorization: Bearer psk_[insert_your_api_token_here' | jq '.[] | .spikeStatus' | grep -c "spike"

Returns 4, as at now... which is 12:51 PM on 07MAY2024

**Reply by mark-leonard-reynolds** [reply]
So, scripting this to run periodically.
Not sure what madpilot thinks about querying frequency, (recommends?) but I think if I make that query once per hour, the load on amber servers is low, and data frequency is good enough for me.
Thinking: Note for beginners : over querying frequency often causes problems.... :-) Especially big queries.

**Reply by madpilot** [reply]
Once a minute is fine (and generally the recommendation), as prices and forecasts change every 5 minutes. If you are happy to run it slower than that, then all good.

**Reply by mark-leonard-reynolds** [reply]
Thanks! My only regret is not doing this sooner! 7pm now, and I can see the aftermath of price volatility,
For an investment of $3.69, which I though was high, I've earnt $32.43 for a net gain of $28,74 for the day.
Enough to pay for my usage for the month. (not all the other fees though, which is my long term aim!)
The spike started a little at 3.30pm, before recovering, and then colapsing again at 5pm to 7pm
![image](https://github.com/amberelectric/public-api/assets/63522018/157791d1-9a19-4ed3-9ed5-8e63fb5866d8)

With corresponding FIT prices.
![image](https://github.com/amberelectric/public-api/assets/63522018/dee0361d-bd98-4a4c-b5a5-74ad3bdcf597)

resulting in this ...
![image](https://github.com/amberelectric/public-api/assets/63522018/0dac219d-2d04-4c46-81b2-33761d5b6a83)

And my own accounting and monitoring
![image](https://github.com/amberelectric/public-api/assets/63522018/6ce68477-442c-4610-b482-6dbd0b8371fd)

**Reply by mark-leonard-reynolds** [reply]
So thinking through a little, system imported 13 kwh during the day, and exported the same 13 kwh.
The buy price was about 25 c/kwh, and the sell price was nearly $1.75. So multiplied by 7.
That's not a bad return on investment. (Ignoring all the many other costs.)
So, I'm guessing this is going to happen more and more in the coming years, so lot's of opportunity
there for amber customers. Yes, my little 13 kwh probably doesn't mean much in the big picture, but 
if thousands of people did the same, it starts to matter.

**Reply by mark-leonard-reynolds** [reply]
I think I'm going to use the same approach to look for very high forecast daily prices, which indicate the same sort
of volatility, which may not trigger the AEMO spike warning. So if there is forecast during the next 24 hours for FIT
going above say $1/kwh, then assume spending 25c/kwh is a good gamble, and load up the battery, even on a
bad solar day. After all, it's the bad solar days (across the whole state) which lead to high FIT prices. Well, one of the
reasons.
Thing is, I've seen lots of high forecast days, where some suppliers seem to swoop in, supply power, and then the
spot prices drop. So hasn't seemed worth the risk in the past. We will see.

**Reply by mark-leonard-reynolds** [reply]
Things are looking like a repeat for tomorrow, according to current AEMO forecast.
https://aemo.com.au/Energy-systems/Electricity/National-Electricity-Market-NEM/Data-NEM/Data-Dashboard-NEM
![image](https://github.com/amberelectric/public-api/assets/63522018/18b77b10-97c6-416b-aefb-dc37409b2a99)
Setting up grid charging for earlier tomorrow I think.
Something seems broken with amber forecast prices around the spike times though. Might resolve as we get closer...
buy
![image](https://github.com/amberelectric/public-api/assets/63522018/2c167b18-6269-45a7-84e4-93b9db965944)

sell
![image](https://github.com/amberelectric/public-api/assets/63522018/81107f55-fbdb-4d96-bc40-5fbbbadba9b3)

**Reply by rob0101** [reply]
You might want to look at the Amber Electric Users Group on Facebook. 

Full of people keen to hear such stories and share outcomes.

**Reply by mark-leonard-reynolds** [reply]
I was wondering where all the amber geeks would be hanging out :-)
But facebook? Really? Think I'd rather talk to myself here :-)

**Reply by rob0101** [reply]
There's also whirlpool. 

https://forums.whirlpool.net.au/thread/9xkzp6q3
